### DISCUSSION TITLE
How to include or exclude merged dictionaries in the apphead.xaml

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#15274] [General]
Hi,

I have a few resources that are only for Windows and i wanted to add them the merged dictionary apphead.xaml, maybe that's the wrong place.

I took a look at the documentation [(https://platform.uno/docs/articles/platform-specific-xaml.html)] and tried to use this as an example, here is what i have tried

```
local:App x:Class="ZiiDMSApp.AppHead"
       xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
       xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
       xmlns:wasm="http://platform.uno/wasm"
       xmlns:local="using:ZiiDMSApp"
       xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
       xmlns:win="using:WinUICommunity"
       mc:Ignorable="wasm win">

    <local:App.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
                <ResourceDictionary Source="ms-appx:///ZiiDMSApp/AppResources.xaml" />

                <win:ResourceDictionary Source="ms-appx:///WinUICommunity.Components/Themes/Generic.xaml"/>
                <win:ResourceDictionary Source="ms-appx:///WinUICommunity.Components/Themes/InfoBarInformationalColor.xaml" />
            </ResourceDictionary.MergedDictionaries>

        </ResourceDictionary>
    </local:App.Resources>

</local:App>
```

This gives me the error
ZiiDMSApp.Shared\AppHead.xaml(16,41): XamlCompiler error WMC0610: The XAML Binary Format (XBF) generator reported syntax error '0x09c6'

open to suggestions

**Comment by jeromelaban** [comment]
This particular error is reported by the WinAppSDK XAML Compiler, a portion that Uno cannot adjust. As a workaround, you can use the `Application.Current.Resources.MergedDictionaries` property and conditional C# to manually add those resources in the `App` constructor.

**Comment by cconner100** [comment]
OK, will not do this in the AppHeader and do it in the code startup as needed. 

Thanks.
